### ISSUE TITLE
skydns matches wrong for its domain when domain names are overlapping. 

### ISSUE THREAD

**Issue by mattthias** [skynetservices/skydns#263]
Hi everyone,

i stumbled over a bug in skydns regarding the matching of its authoritive domain. 

How to reproduce: 

Start a skydns instance with SKYDNS_DOMAIN "ndocode.com". When asking the service to resolve the domain "endocode.com" it does not forwards the query to the authoritative server but answered the query by itself.  

``` bash
sudo ./skydns -verbose=true -domain=ndocode.com
2016/04/12 11:32:35 skydns: falling back to default configuration, could not read from etcd: client: etcd cluster is unavailable or misconfigured
2016/04/12 11:32:35 skydns: metrics enabled on :/metrics
2016/04/12 11:32:35 skydns: ready for queries on ndocode.com. for tcp://127.0.0.1:53 [rcache 0]
2016/04/12 11:32:35 skydns: ready for queries on ndocode.com. for udp://127.0.0.1:53 [rcache 0]
```

``` bash
dig endocode.com @127.0.0.1
; <<>> DiG 9.9.5-9+deb8u6-Debian <<>> endocode.com @127.0.0.1
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 65330
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 0

;; QUESTION SECTION:
;endocode.com.                  IN      A

;; AUTHORITY SECTION:
ndocode.com.            60      IN      SOA     ns.dns.ndocode.com. hostmaster.skydns.local. 1460451600 28800 7200 604800 60

;; Query time: 1 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Tue Apr 12 11:39:29 CEST 2016
;; MSG SIZE  rcvd: 104

```

The problem raises because the function "string.HasSuffix" is used in server.go. Please see my pull request how i tried to fix that problem.

Best wishes,
Matthias

**Comment by arizvisa** [comment]
This can be closed as its' been fixed by PR #264, right?

**Comment by mattthias** [comment]
Hi @arizvisa ,

yes, i think that is fixed :-)
